src/sentry/db/mixin.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import
+
+import logging
+import six
+
+from uuid import uuid4
+
+from sentry.models import OrganizationOption
+
+logger = logging.getLogger('sentry.deletions')
+
+
+def delete_pending_deletion_option(instance, **kwargs):
+    if hasattr(instance, 'delete_pending_deletion_option'):
+        instance.delete_pending_deletion_option()
+
+
+class PendingDeletionMixin(object):
```

Comment:
Do we need an abstraction at this point? With only one implementation it is hard to know that the abstraction is covering a variety of use cases. I think your implementation could be simpler if you focused on solving the problem just for repositories. You wouldn't need to pull in OrganizationOption records then, and in the future when another situation like this comes up an abstraction could be extracted from the two implementations.

Reply:
Yes. As we discussed today, the hope is to use it for other models as well. that may not have a config. I have created a similar pr for the project model.